core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -129,7 +130,7 @@ private DataTask task(StaticTableScan scan) {
                   root.posDeleteFileCount,
                   root.eqDeleteRecordCount,
                   root.eqDeleteFileCount,
-                  root.lastUpdatedMs,
+                  Optional.ofNullable(root.lastUpdatedAt).map(t -> t * 1000).orElse(null),
```

Comment:
Am i missing something here?  Is it just moving the multiply by 1000, from update() to here?

Reply:
yes Szehon, I am not changing how we read from partition metadata table, But the data is stored at microsecond precision instead (if needed to expose via other means)